beacon-chain/db/kv/state.go

```diff
@@ -311,6 +244,113 @@ func (s *Store) saveStatesEfficientInternal(ctx context.Context, tx *bolt.Tx, bl
 	return s.storeValidatorEntriesSeparately(ctx, tx, validatorsEntries)
 }
 
+func savePhase0Efficient(ctx context.Context, rawType interface{}, bucket *bolt.Bucket, blockRoot [32]byte, valIdxBkt *bolt.Bucket, valKey []byte) error {
```

Comment:
I extracted several functions because the complexity of the main func was too big for our CI check